pkg/process/metadata/parser/service.go

```diff
@@ -334,3 +336,22 @@ func parseCommandContextJava(args []string) string {
 
 	return "java"
 }
+
+// parseCommandContextDotnet extracts metadata from a dotnet launcher command line
+func parseCommandContextDotnet(args []string) string {
+	for _, a := range args {
+		if strings.HasPrefix(a, "-") {
+			continue
+		}
+		// when running assembly's dll, the cli must be executed without command
+		// https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-run#description
+		if strings.HasSuffix(strings.ToLower(a), ".dll") {
+			_, file := filepath.Split(a)
+			return file[:len(file)-4]
+		}
+		// dotnet cli syntax is something like `dotnet <cmd> <args> <dll> <prog args>`
+		// if the first non arg (`-v, --something, ...) is not a dll file, exit early since nothing is matching a dll execute case
+		break
+	}
+	return "dotnet"
+}
```

Comment:
```suggestion

const (
	dllSuffix = ".dll"
)

// parseCommandContextDotnet extracts metadata from a dotnet launcher command line
func parseCommandContextDotnet(args []string) string {
	for _, a := range args {
		if strings.HasPrefix(a, "-") {
			continue
		}
		// when running assembly's dll, the cli must be executed without command
		// https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-run#description
		if strings.HasSuffix(strings.ToLower(a), dllSuffix) {
			_, file := filepath.Split(a)
			return file[:len(file)-len(dllSuffix)]
		}
		// dotnet cli syntax is something like `dotnet <cmd> <args> <dll> <prog args>`
		// if the first non arg (`-v, --something, ...) is not a dll file, exit early since nothing is matching a dll execute case
		break
	}
	return "dotnet"
}
```